File: binaryToDecimal.java
package springIntro;

import java.util.*;

public class binaryToDecimal {

	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		System.out.println("Enter the binary number");
		int binary = in.nextInt();
		int pow = 1, decimal = 0;
		while(binary > 0) {
			decimal += (binary % 10) * pow;
			pow *= 2;
			binary /= 10;
		}
		System.out.println("Decimal equivalent = " + decimal);
	}

}

File: binaryToDecimal.cpp
#include<iostream>
#include<cmath>
using namespace std;
int main(){
	int binary, decimal = 0, counter = 0;
	int size = 0;
	cout << "Enter a binary number = ";
	cin >> binary;
	size = sizeof(binary);
	for(int i = 0; i < size; i++){
		if(binary % 10 == 1){
			decimal += pow(2, counter);
		} 
		counter++;
		binary /= 10;
	}
	cout << "Decimal equivalent = " << decimal;
	return 0;
}
